lib/domain.gi

```diff
@@ -37,6 +40,9 @@ InstallMethod( \=,
     IsIdenticalObj,
     [ IsDomain, IsCollection and IsList ],
     function( D, C )
+    if HasIsFinite(C) and HasIsFinite(D) and (IsFinite(C) <> IsFinite(D)) then
+      return false;
+    fi;
```

Comment:
Just some ramblings:

An alternative to this change would be this:
```gap
InstallMethod( \=, [HasIsFinite, HasIsFinite], 10,
function(x, y)
  if IsFinite(x) <> IsFinite(y) then return false; fi;
  TryNextMethod();
end);
```

Thinking some more about this: since this is a fallback method *and* we are going to enumerate `C` anyway, we could drop the two `HasIsFinite` checks, and just ask for `IsFinite` -- I am pretty sure that if `IsFinite(C)` fails to terminate, then so will `IsSSortedList(C)` (at least for any "sane" object).